src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -5625,60 +5641,158 @@ namespace SurfaceGeometry {
+                    SurfaceTmp(SurfNum).SinAzim = 0.0; // Tuned Was std::sin( 0.0 )
+                    // Outward normal unit vector (pointing away from room)
+                    SurfaceTmp(SurfNum).OutNormVec = SurfaceTmp(SurfNum).lcsz;
+                    SurfaceTmp(SurfNum).ViewFactorSky = 0.5;
+                    SurfaceTmp(SurfNum).ExtSolar = false;
+                    SurfaceTmp(SurfNum).ExtWind = false;
+                    SurfaceTmp(SurfNum).BaseSurf = SurfNum;
+                    SurfaceTmp(SurfNum).BaseSurfName = SurfaceTmp(SurfNum).Name;
+                    SurfaceTmp(SurfNum).ExtBoundCondName = SurfaceTmp(SurfNum).Name;
+                    SurfaceTmp(SurfNum).ExtBoundCond = UnreconciledZoneSurface;
+                }
             }
-            SurfaceTmp(SurfNum).GrossArea = rNumericArgs(1);
-            SurfaceTmp(SurfNum).Area = SurfaceTmp(SurfNum).GrossArea;
-            SurfaceTmp(SurfNum).NetAreaShadowCalc = SurfaceTmp(SurfNum).Area;
-            SurfaceTmp(SurfNum).Width = SurfaceTmp(SurfNum).Area;
-            SurfaceTmp(SurfNum).Height = 1.0;
-            SurfaceTmp(SurfNum).Tilt = 90.0;
-            SurfaceTmp(SurfNum).CosTilt = 0.0; // Tuned Was std::cos( 90.0 * DegToRadians )
-            SurfaceTmp(SurfNum).SinTilt = 1.0; // Tuned Was std::sin( 90.0 * DegToRadians )
-            SurfaceTmp(SurfNum).Azimuth = 0.0;
-            SurfaceTmp(SurfNum).CosAzim = 1.0; // Tuned Was std::cos( 0.0 )
-            SurfaceTmp(SurfNum).SinAzim = 0.0; // Tuned Was std::sin( 0.0 )
-            // Outward normal unit vector (pointing away from room)
-            SurfaceTmp(SurfNum).OutNormVec = SurfaceTmp(SurfNum).lcsz;
-            SurfaceTmp(SurfNum).ViewFactorSky = 0.5;
-            SurfaceTmp(SurfNum).ExtSolar = false;
-            SurfaceTmp(SurfNum).ExtWind = false;
-            SurfaceTmp(SurfNum).BaseSurf = SurfNum;
-            SurfaceTmp(SurfNum).BaseSurfName = SurfaceTmp(SurfNum).Name;
-            SurfaceTmp(SurfNum).ExtBoundCondName = SurfaceTmp(SurfNum).Name;
-            SurfaceTmp(SurfNum).ExtBoundCond = UnreconciledZoneSurface;
         }
     }
 
+    int GetNumIntMassSurfaces(int const TotIntMass // Number of Internal Mass Surfaces to obtain
+    )
+    {
+        // Counts internal mass surfaces applied to zones and zone lists
+
+        // Using/Aliasing
+        using namespace DataIPShortCuts;
+
+        int IOStat;          // IO Status when calling get input subroutine
+        int SurfaceNumAlpha; // Number of material alpha names being passed
+        int SurfaceNumArg;   // Number of material properties being passed
+        int NumIntMassSurf;  // total count of internal mass surfaces
+
+        NumIntMassSurf = 0;
+        if (TotIntMass == 0) return NumIntMassSurf;
+
+        cCurrentModuleObject = "InternalMass";
+        // scan for zones and zone lists in InternalMass objects
+        for (int Item = 1; Item <= TotIntMass; ++Item) {
```

Comment:
It's interesting that this function is just looping up to a specified internal mass counter index.  I'll have to understand the logic from the code calling it...

Reply:
Oh I see, it's just looping over all of them.  OK, it may be clearer to just get GetNumObjects from within this function so you don't have to pass the argument in.  I was confused because I thought this function was called multiple times, one for each mass index, and then it would give you the count **up to** that mass or something.

Reply:
OK. I will use GetNumObjects function instead of passing the "TotIntMass" aregument. But I will have to do it in two different places.